### ISSUE TITLE
quoted-string should allow "optional one-type-only quotes"

### ISSUE THREAD

**Issue by drAlberT** [adrienverge/yamllint#418]
The following does not produce the expected result:

```
  quoted-strings:
    quote-type: single
    required: false
```

The "required: false" part is superseded by the presence of the "single" quote type ...

This should not happen, because it is legit to limit the intended behaviour to  "_**allow only single quotes, or no quotes at all**_".

**Comment by adrienverge** [comment]
Hello,

I don't understand your question. Can you provide an example of YAML file, what you expect to be reported, and what you see?

Additionally, please check out what is expected with `quote-type: single, required: false`: https://github.com/adrienverge/yamllint/blob/9e6dfac/tests/rules/test_quoted_strings.py#L201-L238
And the documentation: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings

**Comment by drAlberT** [comment]
Hi,
  I need to use single quotes when a quote is used, without forcing to have quoting only when needed.

The user is free to quote or not the string, but once he chose to quote then he has to use single quotes.

**Comment by drAlberT** [comment]
```
not_required: bar
not_required_but_allowed: 'foo'
required: '*123'
required_but_wrong: "*afgf"  # FAIL
```

@adrienverge the above example could clarify my intent I hope .. 

I want to force the quote type, without enforcing every string to be quoted.

**Comment by adrienverge** [comment]
No it does not clarify your problem, and takes time to read and try to understand you.
- Again: what do you expect to be reported? What output you see? Can you provide a reproduction scenario (including commands and files used)?
- Again, please check out how yamllint behaves with `quote-type: single, required: false`: https://github.com/adrienverge/yamllint/blob/9e6dfac/tests/rules/test_quoted_strings.py#L201-L238
  Is it different from your expectation "*I want to force the quote type, without enforcing every string to be quoted.*"?

**Comment by drAlberT** [comment]
Ok never mind, if you don't understand this simple sentence I can't do anything.

Again, I read the fm of course.

**Comment by adrienverge** [comment]
I understand English but the problem you describe doesn't exist:
- when trying on my computer,
- in automated tests that yamllint runs on every commit: https://github.com/adrienverge/yamllint/blob/9e6dfac/tests/rules/test_quoted_strings.py#L201-L238
   ```python
  conf = 'quoted-strings: {quote-type: single, required: false}\n'
  self.check('---\n'
             'string1: foo\n'                          # doesn't fail
             'string2: "foo"\n'                        # fails
             'string5: \'foo\'\n'                      # doesn't fail
             ...
  ```
